ogc/query-with-cql-filters/src/main/java/com/esri/samples/query_with_cql_filters/QueryWithCqlFiltersController.java

```diff
@@ -0,0 +1,316 @@
+import com.esri.arcgisruntime.symbology.SimpleRenderer;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Button;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.DatePicker;
+import javafx.scene.control.Label;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ProgressIndicator;
+import javafx.scene.control.TextField;
+import javafx.scene.paint.Color;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+public class QueryWithCqlFiltersController {
+
+  @FXML private Button applyQueryButton;
+  @FXML private Button revertToInitialQueryButton;
+  @FXML private CheckBox timeExtentCheckBox;
+  @FXML private ComboBox<String> comboBox;
+  @FXML private DatePicker endDatePicker;
+  @FXML private DatePicker startDatePicker;
+  @FXML private Label featureNumberLabel;
+  @FXML private MapView mapView;
+  @FXML private ProgressIndicator progressIndicator;
+  @FXML private TextField textField;
+
+  private Envelope datasetExtent;
+  private OgcFeatureCollectionTable ogcFeatureCollectionTable; // keep loadable in scope to avoid garbage collection
+
+  public void initialize() {
+
+    try {
+
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create a map with the topographic basemap style
+      ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC);
+
+      // set the map to the mapview
+      mapView.setMap(map);
+
+      // prepare UI for interaction
+      populateUiWithCqlQueriesAndDateValues();
+
+      // define strings for the service URL and collection id
+      // note that the service defines the collection id which can be accessed via OgcFeatureCollectionInfo.getCollectionId().
+      String serviceUrl = "https://demo.ldproxy.net/daraa";
+      String collectionId = "TransportationGroundCrv";
+
+      // create an OGC feature collection table from the service url and collection id
+      ogcFeatureCollectionTable = new OgcFeatureCollectionTable(serviceUrl, collectionId);
+
+      // set the feature request mode to manual
+      // in this mode, the table must be manually populated - panning and zooming won't request features automatically
+      ogcFeatureCollectionTable.setFeatureRequestMode(ServiceFeatureTable.FeatureRequestMode.MANUAL_CACHE);
+
+      // load the table
+      ogcFeatureCollectionTable.loadAsync();
+
+      // ensure the feature collection table has loaded successfully before creating a feature layer from it to display on the map
+      ogcFeatureCollectionTable.addDoneLoadingListener(() -> {
```

Comment:
You can probably just create the layer and listen to any error from that since the layer will automatically load the table and the layer will automatically load when the map itself loads. The original intention as far as I know was that loadables would load automatically.